From 7d297494b7e8f413fee67f3a2e86b33c9c5d5ffb Mon Sep 17 00:00:00 2001
From: Paul Smith <psmith@gnu.org>
Date: Sun, 20 Oct 2013 12:49:01 -0400
Subject: [PATCH 1/2] * read.c (record_files): [SV 33034] Change fatal() to
 error()

Allows deprecated syntax.  However we don't guarantee this syntax
will continue to be legal in the future.
Change suggested by David Boyce <david.s.boyce@gmail.com>
---
 src/kmk/NEWS   | 10 ++++++++--
 src/kmk/read.c |  2 +-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/kmk/NEWS b/src/kmk/NEWS
index 523c9a7..c3e0c8b 100644
--- a/src/kmk/NEWS
+++ b/src/kmk/NEWS
@@ -114,6 +114,12 @@ http://sv.gnu.org/bugs/index.php?group=make&report_id=111&fix_release_id=104&set
   after the variable name, to allow for simple, conditional, or appending
   multi-line variable assignment.
 
+* Change the fatal error for mixed explicit and implicit rules, that was
+  introduced in GNU make 3.82, to a non-fatal error.  However, this syntax is
+  still deprecated and may return to being illegal in a future version of GNU
+  make.  Makefiles that rely on this syntax should be fixed.
+  See https://savannah.gnu.org/bugs/?33034
+
 
 Version 3.81
 
@@ -128,8 +134,8 @@ Version 3.81
   any prerequisite that does not exist, even though that prerequisite
   might have caused the target to rebuild.  Starting with the _next_
   release of GNU make, '$?' will contain all prerequisites that caused
-  the target to be considered out of date.  See this Savannah bug:
-  http://savannah.gnu.org/bugs/index.php?func=detailitem&item_id=16051
+  the target to be considered out of date.
+  See http://savannah.gnu.org/bugs/?16051
 
 * WARNING: Backward-incompatibility!
   GNU make now implements a generic "second expansion" feature on the
diff --git a/src/kmk/read.c b/src/kmk/read.c
index ebc6a63..9a326bb 100644
--- a/src/kmk/read.c
+++ b/src/kmk/read.c
@@ -2746,7 +2746,7 @@ l_next:
       /* Reduce escaped percents.  If there are any unescaped it's an error  */
       name = filenames->name;
       if (find_percent_cached (&name))
-        fatal (flocp, _("mixed implicit and normal rules"));
+        error (flocp, _("*** mixed implicit and normal rules: deprecated syntax"));
     }
 }
 
-- 
2.1.1

